src/target/ch32f1.c

```diff
@@ -139,11 +139,14 @@ static bool ch32f1_flash_unlock(target *t)
 	uint32_t cr = target_mem_read32(t, FLASH_CR);
 	if (cr & FLASH_CR_FLOCK_CH32) {
```

Comment:
As part of doing this cleanup, it would be a good idea to turn this into something more like:
```c
if (cr & FLASH_CR_FLOCK_CH32)
    DEBUG_WARN("Fast unlock failed, cr: 0x%08" PRIx32 "\n", cr);
return !(cr & FLASH_CR_FLOCK_CH32);
```

We should prefer returning a value that's coerced to bool rather than explicit bool values.

Reply:
This still has the extra {}'s that should be removed, but we'll accept this as-is as that can be fixed on the target cleanup branch